app/services/qa/linked_data/graph_service.rb

```diff
@@ -42,17 +41,39 @@ def object_values(graph:, subject:, predicate:)
           values
         end
 
+        def deep_copy(graph:)
+          new_graph = RDF::Graph.new
+          graph.statements.each do |st|
+            new_graph.insert(st.dup)
+          end
+        end
+
         private
 
           def filter_out_blanknode(remove, subj)
             remove && subj.anonymous?
           end
 
-          def filter_out_language(language, obj)
+          # Filter out language based on...
+          # * do not remove if the object literal does not respond to :language
+          # * do not remove if the object literal does not have a language marker
+          # * do not remove if the object has the targeted language marker
+          # * do not remove if none of the other objects with this statement's predicate have the targeted language marker
+          def filter_out_language(graph, language, statement)
             return false if language.blank?
-            return false unless obj.respond_to?(:language)
-            return false if obj.language.blank?
-            !language.include?(obj.language)
+            return false unless Qa::LinkedData::LanguageService.literal_has_language_marker?(statement.object)
+            objects = object_values(graph: graph, subject: statement.subject, predicate: statement.predicate)
+            return false unless at_least_one_object_has_language?(objects, language)
+            !language.include?(statement.object.language)
+          end
```

Comment:
If the external authority filtered out to a language other than the preferred language, the previous filtering process could return 0 results for a predicate.  The adjustment checks to see if there are any values from other statements for a predicate that will remain in the graph (e.g. at least one that has the preferred language). If not, all values for the predicate are kept in the graph.  

This makes more sense than returning 0 statements for a predicate that may actually have multiple values.